lib/nsxv.py

```diff
@@ -238,12 +242,29 @@ def __str__(self):
           destination_address = nacaddr.ExcludeAddrs(
               destination_address,
               destination_address_exclude)
-        if not destination_address:
-          logging.warn(self.NO_AF_LOG_ADDR.substitute(term=self.term.name,
-                                                      direction='destination',
-                                                      af=self.filter_type))
+
+        if destination_address:
+          if af == 4:
+            dest_ipv4_address = True
+          else:
+            dest_ipv6_address = True
         destination_addr.extend(destination_address)
 
+    # Drop the mismatch IP if source and destination address are not valid
+    # for mixed filter
+    if self.filter_type == 'mixed':
+      if source_addr and destination_addr:
+        if source_ipv4_address and not dest_ipv4_address:
+          source_addr = self._DropMismatchIpAddress(source_addr, 'source', 4)
+        elif source_ipv6_address and not dest_ipv6_address:
+          source_addr = self._DropMismatchIpAddress(source_addr, 'source', 6)
+        elif dest_ipv4_address and not source_ipv4_address:
+          destination_addr = self._DropMismatchIpAddress(destination_addr,
+                                                 'destination', 4)
+        elif dest_ipv6_address and not source_ipv6_address:
+          destination_addr = self._DropMismatchIpAddress(destination_addr,
+                                                 'destination', 6)
```

Comment:
One use case that I'm not sure if the correct behavior happened is:

v4-to-v6 (and vice versa)
it currently drops the source address and keeps the destination address. I'm not sure what the best behavior would be. It feels like the generator gets to a situation where it removes all the address for a side and thus making it that side any, that a warning or maybe even a dropping of the whole term is warranted. But i'm not sure. What is your take on it?
@ankenyr  what do you think should happen in this situation (lets say from an ideal perspective even if it's not what other generators do)